renderer/backend/metal/command_buffer_mtl.mm

```diff
@@ -173,6 +173,7 @@ static void LogMTLCommandBufferErrorIfPresent(id<MTLCommandBuffer> buffer) {
   }
 
   [buffer_ commit];
+  [buffer_ waitUntilScheduled];
```

Comment:
Can we write a test for this that would document what can go wrong here?

If not, can you add some comment explaining what can go wrong if this gets messed up/dropped/moved to the wrong spot?

Reply:
If we messed this up, we would run into the GPU hangs that were reported in existing tests. This is definitely not safe to encoding subpass contents in parallel. @bdero and I were in agreement that we would need to expose explicit synchornization mechanisms for that. That is easy to do in Metal & Vulkan but we don't know what the implications are going to be for OpenGL ES. We won't be performing parallel encoding on that backend, but I don't want us to back us into a corner with an API that provides synchonization mechanisms out of the box.